storage-node/src/services/webApi/app.ts

```diff
@@ -73,6 +74,16 @@ export async function createApp(config: AppConfig): Promise<Express> {
       }
     },
 
+    // Pre validate data object id in GET/HEAD file requests
+    (req: express.Request, res: express.Response<unknown, AppConfig>, next: NextFunction) => {
+      if (req.path.startsWith('/api/v1/files/') && (req.method === 'GET' || req.method === 'HEAD')) {
+        validateDataObjectId(req, res)
+          .then(next)
+          .catch((error) => sendResponseWithError(res, next, error, 'upload'))
+      } else {
+        next()
+      }
+    },
```

Comment:
Actually, this still won't work correctly as `req.params` inside `validateDataObjectId` is empty object (`{}`). This is because the express does not know about the structure of the URL yet (i.e. whether  `1` in `/api/v1/files/1` is parameter or part of the URL itself), and it will only know about the `params` after the next middleware (`OpenApiValidator.middleware`) is executed.

So unfortunately I think there is no way then doing this validation at openapi schema level

Reply:
Ah thanks for checking this. 
So I'll drop it, and update the handlers to not assume the id has been validated fully, and at least update the schema validation.

Reply:
Done in [0cdd273](https://github.com/Joystream/joystream/pull/5016/commits/0cdd2732bb3333040b16b714a081aeb46d8a8bc5)